src/structures/CommandInteractionOptionResolver.js

```diff
@@ -0,0 +1,159 @@
+'use strict';
+
+/**
+ * A resolver for command interaction options.
+ */
+class CommandInteractionOptionResolver {
+  constructor(options) {
+    /**
+     * The interaction options array.
+     * @type {CommandInteractionOption[]}
+     * @private
+     */
+    this.options = options ?? [];
+  }
+
+  /**
+   * Gets an option by its name.
+   * @param {string} name The name of the option.
+   * @param {boolean} required Whether to throw an error if the option is not found.
+   * @returns {?CommandInteractionOption} The option, if found.
+   */
+  get(name, required = false) {
+    const option = this.options.find(opt => opt.name === name);
+    if (option) {
+      return option;
+    } else if (required) {
+      throw new Error(`Missing required option "${name}"`);
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Gets an option by name and property and checks its type.
+   * @param {string} name The name of the option.
+   * @param {ApplicationCommandOptionType[]} types The type of the option.
+   * @param {string[]} properties The property to check for for `required`.
+   * @param {boolean} required Whether to throw an error if the option is not found.
+   * @returns {*|null} The value of `property`.
+   * @private
+   */
+  _getTypedOption(name, types, properties, required) {
+    const option = this.get(name, required);
+    if (option) {
+      if (!types.includes(option.type)) {
+        throw new Error(`Option "${name}" is of type "${option.type}"; expected ${types.join('|')}.`);
```

Comment:
Some users might not be familiar with the `|` notation
```suggestion
        throw new Error(`Option "${name}" is of type "${option.type}"; expected one of: ${types.join(', ')}.`);
```